src/pages/ListOfMovies/ListOfMovies.tsx

```diff
@@ -0,0 +1,109 @@
+import { Box, Grid, Pagination, PaginationItem, Typography } from '@mui/material';
+import { createListOfMoviesStyles } from './createListOfMoviesStyles';
+import { FilterButton } from '../../UIKit/FilterButton/FilterButton';
+import { ChangeEvent, useContext, useEffect, useState } from 'react';
+import { DECADES, GENRES, SORT_BY } from './filterConstants';
+import { IResponseList } from '../../api/apiTypes/apiTypes';
+import { IMovieDiscover } from '../../types/movieTypes';
+import { getMovieList } from '../../api/api';
+import { ImageConfig } from '../../providers/ImageConfigProvider/ImageConfigContexts';
+import { API_MOVIE_LIST_PARAMS, imageSizes, SORT_BY_OPTIONS } from '../../constants';
+import { Link as RouterLink } from 'react-router-dom';
+
+const styles = createListOfMoviesStyles();
+
+export const ListOfMovies = () => {
+  const [movieList, setMovieList] = useState<IResponseList<IMovieDiscover[]>>();
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [activeGenre, setActiveGenre] = useState(GENRES[0]);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [activeDecade, setActiveDecade] = useState(DECADES[0]);
+  const [activeSortBy, setActiveSortBy] = useState<SORT_BY_OPTIONS>(SORT_BY[0].value);
+
+  const [page, setPage] = useState(1);
+
+  const handleChange = (event: ChangeEvent, value: number) => {
+    setPage(value);
+  };
+
+  useEffect(() => {
+    getMovieList({
+      params: {
+        [API_MOVIE_LIST_PARAMS.PAGE]: page,
+        [API_MOVIE_LIST_PARAMS.SORT_BY]: activeSortBy,
+      },
+    }).then((data) => setMovieList(data));
+  }, [page, activeSortBy]);
+
+  const Header = () => {
+    const handleChangeGenre = (genre: string) => {
+      setActiveGenre(genre);
+    };
+
+    const handleChangeDecade = (decade: string) => {
+      setActiveDecade(decade);
+    };
+
+    const handleChangeSort = (sort: SORT_BY_OPTIONS) => {
+      setActiveSortBy(sort);
+    };
+
+    return (
+      <Grid item sx={styles.itemWrapperStyles}>
+        <Typography sx={styles.headerTypographyStyles}>FILMS</Typography>
+        <Box display="flex" justifyContent="center">
+          <FilterButton onChange={handleChangeGenre} item={GENRES} />
+          <FilterButton onChange={handleChangeDecade} item={DECADES} />
+          <FilterButton onChange={handleChangeSort} item={SORT_BY.map((s) => s.label)} />
+        </Box>
+      </Grid>
+    );
+  };
+
+  const MovieBox = ({ movieList }: { movieList?: IMovieDiscover[] }) => {
+    const imageConfig = useContext(ImageConfig);
+
+    return (
+      <Grid item container p={1} display="flex" justifyContent="space-between">
+        {movieList?.map((movie) => (
+          <Grid
+            item
+            key={movie.id}
+            display="flex"
+            justifyContent="center"
+            sx={styles.posterImgStyles}
```

Comment:
The remaining properties can also be moved to styles

Reply:
ok